presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -185,13 +194,28 @@ private static boolean isColumnReference(QualifiedName name, RelationType relati
         return Optional.empty();
     }
 
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .addValue(relationId)
+                .toString();
+    }
+
     public static final class Builder
     {
+        private RelationId relationId = RelationId.anonymous();
```

Comment:
I did so, but there are places outside of analyzer which do not benefit from `RelationId` requirement (planner, local planner) and still pay its cost. I'm not sure the net result is in plus.